schemas/6.json

```diff
@@ -0,0 +1,1121 @@
+      "title": "Rule Options",
+      "allOf": [
+        {
+          "if": {
+            "properties": {
+              "operator": {
+                "enum": [
+                  "EXACTLY",
+                  "NOT",
+                  "GREATER_THAN",
+                  "GREATER_THAN_OR_EQUAL",
+                  "LESS_THAN",
+                  "LESS_THAN_OR_EQUAL",
+                  "INCLUDES",
+                  "EXCLUDES"
+                ]
+              }
+            }
+          },
+          "then": {
+            "required": [
+              "value"
+            ]
+          }
+        }
+      ]
+    },
+    "Prompt": {
+      "type": "object",
+      "additionalProperties": false,
+      "properties": {
+        "id": {
+          "type": "string"
+        },
+        "text": {
+          "type": "string"
+        },
+        "additionalAttributes": {
+          "$ref": "#/definitions/AdditionalAttributes"
+        },
+        "variable": {
+          "type": "string"
+        },
+        "edgeVariable": {
+          "type": "string"
+        },
+        "negativeLabel": {
+          "type": "string"
+        },
+        "otherVariable": {
+          "type": "string"
+        },
+        "otherVariablePrompt": {
+          "type": "string"
+        },
+        "otherOptionLabel": {
+          "type": "string"
+        },
+        "bucketSortOrder": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/SortOrder"
+          }
+        },
+        "binSortOrder": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/SortOrder"
+          }
+        },
+        "sortOrder": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/SortOrder"
+          }
+        },
+        "color": {
+          "type": "string"
+        },
+        "layout": {
+          "$ref": "#/definitions/Layout"
+        },
+        "edges": {
+          "$ref": "#/definitions/Edges"
+        },
+        "highlight": {
+          "$ref": "#/definitions/Highlight"
+        },
+        "createEdge": {
+          "type": "string"
+        }
+      },
+      "required": [
+        "id",
+        "text"
+      ],
+      "title": "Prompt"
+    },
+    "Preset": {
+      "type": "object",
+      "additionalProperties": false,
+      "properties": {
+        "id": {
+          "type": "string"
+        },
+        "label": {
+          "type": "string"
+        },
+        "layoutVariable": {
+          "type": "string"
+        },
+        "groupVariable": {
+          "type": "string"
+        },
+        "edges": {
+          "$ref": "#/definitions/Edges"
+        },
+        "highlight": {
+          "$ref": "#/definitions/NarrativeHighlight"
+        }
+      },
+      "required": [
+        "id",
+        "label",
+        "layoutVariable"
+      ],
+      "title": "Preset"
+    },
+    "Behaviours": {
+      "type": "object",
+      "additionalProperties": false,
```

Comment:
I'm thinking perhaps we should set this to `true` so that future behaviour implementations don't require a new schema version. What do you think?

Reply:
Hmm, yes it does seem a bit restrictive.

I think so long as interfaces can work without properties being defined then that would be okay.

Maybe Interviewer could flag unknown additonal properties?

Reply:
Yeah, I think that's a better way of doing it.